libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_common.c

```diff
@@ -1061,6 +1065,46 @@ void IotTestBleHal_BTUnregister( void )
     TEST_ASSERT_EQUAL( eBTStatusSuccess, xStatus );
 }
 
+void IotTestBleHal_CreateSecureConnection_Model1Level4( bool IsBondSucc )
+{
+    BTStatus_t xStatus;
+    BLETESTsspRequestCallback_t xSSPrequestEvent;
+    BLETESTPairingStateChangedCallback_t xPairingStateChangedEvent;
+
+    /* Wait secure connection. Secure connection is triggered by writting to bletestsCHARB. */
+    xStatus = IotTestBleHal_WaitEventFromQueue( eBLEHALEventSSPrequestCb, NO_HANDLE, ( void * ) &xSSPrequestEvent, sizeof( BLETESTsspRequestCallback_t ), BLE_TESTS_WAIT );
+    TEST_ASSERT_EQUAL( eBTStatusSuccess, xStatus );
+    TEST_ASSERT_EQUAL( 0, memcmp( &xSSPrequestEvent.xRemoteBdAddr, &_xAddressConnectedDevice, sizeof( BTBdaddr_t ) ) );
+    TEST_ASSERT_EQUAL( eBTsspVariantConsent, xSSPrequestEvent.xPairingVariant );
+
+    xStatus = _pxBTInterface->pxSspReply( &xSSPrequestEvent.xRemoteBdAddr, eBTsspVariantConsent, true, 0 );
+    TEST_ASSERT_EQUAL( eBTStatusSuccess, xStatus );
+
+    xStatus = IotTestBleHal_WaitEventFromQueue( eBLEHALEventSSPrequestConfirmationCb, NO_HANDLE, ( void * ) &xSSPrequestEvent, sizeof( BLETESTsspRequestCallback_t ), BLE_TESTS_WAIT );
+    TEST_ASSERT_EQUAL( eBTStatusSuccess, xStatus );
```

Comment:
Are all the pairing events guaranteed to be received when pairing is cancelled ?

Reply:
disable callback check